app/src/main/java/com/android/solvit/seeker/ui/provider/ProviderInfo.kt

```diff
@@ -0,0 +1,401 @@
+                  text = "CHF ${provider.price}/hour",
+                  fontSize = 16.sp,
+                  color = Color.Gray,
+                  modifier = Modifier.padding(vertical = 4.dp).testTag("priceDisplay"))
+            }
+
+        Spacer(modifier = Modifier.height(16.dp))
+
+        Text(
+            text = "Descriptions",
+            fontSize = 18.sp,
+            fontWeight = FontWeight.Bold,
+            modifier = Modifier.testTag("descriptionTitle"))
+
+        Spacer(modifier = Modifier.height(8.dp))
+
+        // Description section with white background and border
+        Column(
+            modifier =
+                Modifier.fillMaxWidth()
+                    .background(color = Color.White, shape = RoundedCornerShape(16.dp))
+                    .border(
+                        width = 2.dp, color = Color.Transparent, shape = RoundedCornerShape(16.dp))
+                    .padding(16.dp)
+                    .testTag("descriptionSection")) {
+              Text(
+                  text = provider.description,
+                  modifier = Modifier.padding(vertical = 4.dp).testTag("descriptionText"),
+                  color = Color.Gray,
+                  fontWeight = FontWeight.Medium)
+            }
+
+        Spacer(modifier = Modifier.height(16.dp))
+
+        Text(
+            text = "Contact",
+            fontSize = 18.sp,
+            fontWeight = FontWeight.Bold,
+            modifier = Modifier.testTag("contactTitle"))
+
+        Spacer(modifier = Modifier.height(8.dp))
+
+        // Contact section with white background and border
+        Column(
+            modifier =
+                Modifier.fillMaxWidth()
+                    .background(color = Color.White, shape = RoundedCornerShape(16.dp))
+                    .border(
+                        width = 2.dp, color = Color.Transparent, shape = RoundedCornerShape(16.dp))
+                    .padding(16.dp)
+                    .testTag("contactSection")) {
+              Row(
+                  verticalAlignment = Alignment.CenterVertically,
+                  horizontalArrangement = Arrangement.spacedBy(16.dp)) {
+                    Icon(
+                        painter = painterResource(id = R.drawable.message),
+                        contentDescription = "Message",
+                        modifier = Modifier.size(64.dp).testTag("messageIcon"),
+                        tint = Color.Unspecified)
+                    Text(
+                        text =
+                            "The contractor's contacts are visible only to its customers." +
+                                "If you are interested in the services of this contractor" +
+                                " - offer him an order.",
+                        modifier = Modifier.padding(vertical = 4.dp).testTag("contactText"),
+                        color = Color.Gray,
+                        fontWeight = FontWeight.Medium)
+                  }
+            }
```

Comment:
Nit: There is quite some duplication between these two rubrics of the Home screen, as well as the header of the "Reviews" rubric in the "Reviews" tab. Would it make sense to introduce an object Rubric in which the content is passed as an argument? This could help keep the conformity and make general UI editing easier. However, for just 3 parts it may be overkill, see my general comment for this file for more details.

Reply:
I implemented this change in my latest commit. Thanks!